crates/miden-testing/tests/integration/scripts/in_flight_swap.rs

```diff
@@ -0,0 +1,239 @@
+use miden_crypto::{Word, rand::FeltRng};
+use miden_lib::{
+    account::wallets::{AuxWallet, BasicWallet},
+    note::utils::{build_p2id_recipient, build_swap_tag},
+    transaction::TransactionKernel,
+};
+use miden_objects::{
+    Felt,
+    account::{Account, AccountId},
+    asset::Asset,
+    crypto::rand::RpoRandomCoin,
+    note::{
+        Note, NoteAssets, NoteDetails, NoteExecutionHint, NoteInputs, NoteMetadata, NoteRecipient,
+        NoteScript, NoteTag, NoteType,
+    },
+    transaction::OutputNote,
+};
+use miden_testing::{AccountState, Auth, MockChain};
+use rand::random;
+
+#[test]
+fn test_inflight_swap() {
```

Comment:
Nit: For new tests my preference is to avoid using unwraps/expects and use `anyhow::Result<()>` instead. That is one way to dogfood our own error messages and see if they're any good when a test fails. With unwrapping, we only get debug representations of errors and miss the full error chain.

I guess this test is probably based off an existing test that used unwraps, so also fine to leave it as is. Just wanted to take this opportunity to mention the general idea of using `anyhow::Result`.